examples/Components/Typeahead/Typeahead.purs

```diff
@@ -2,178 +2,198 @@ module Docs.Components.Typeahead where
+      , HH.slot _select unit (Select.component handleExtraActions) input (Just <<< HandleSelect)
+      ]
+    where
+    input =
+      { initialSearch: Nothing
+      , debounceTime: Nothing
+      , inputType: Select.TextInput
+      , items: difference st.items st.selected
+      , render: renderSelect
+      }
+
+  handleExtraActions 
+    :: Variant ExtraActions 
+    -> H.HalogenM _ (Select.Action TypeaheadItem ExtraActions () m) () _ m Unit
+  handleExtraActions = match  
+    { log: \str -> Console.log str 
+    }
+
+  handleAction :: Action -> H.HalogenM State Action (ChildSlots m) Message m Unit
+  handleAction = case _ of
+    HandleSelect msg -> match
+      { searched: \search -> do
           st <- H.get
           let newItems = difference (filterItems search st.items) st.selected
               index = elemIndex search st.items
-          _ <- H.query unit $ Select.replaceItems newItems
-          traverse_ (H.query unit <<< Select.highlight <<< Select.Index) index
-
-        Select.Selected item -> do
+          _ <- H.query _select unit $ H.tell $ Select.ReplaceItems newItems
+          for_ index \ix -> 
+            H.query _select unit $ H.tell $ Select.PerformAction $ Select.highlight $ Select.Index ix
+      
+      , selected: \item -> do
           st <- H.get
 
-          _ <- if st.keepOpen
-            then pure unit
-            else do
-              _ <- H.query unit $ Select.setVisibility Select.Off
-              pure unit
+          when (not st.keepOpen) do
+            void $ H.query _select unit $ H.tell $ Select.PerformAction $ Select.setVisibility Select.Off
 
-          if length (filter ((==) item) st.items) > 0
-            then H.modify_ _ { selected = ( item : st.selected ) }
-            else H.modify_ _
-                  { items = ( item : st.items )
-                  , selected = ( item : st.selected ) }
+          case length (filter ((==) item) st.items) of
+            0 ->
+              H.modify_ _ { items = ( item : st.items ), selected = ( item : st.selected ) }
+            _ ->
+              H.modify_ _ { selected = ( item : st.selected ) }
 
           newSt <- H.get
           let newItems = difference newSt.items newSt.selected
-          _ <- H.query unit $ Select.replaceItems newItems
-          pure unit
-
-        otherwise -> pure unit
+          void $ H.query _select unit $ H.tell $ Select.ReplaceItems newItems
+      
+      , visibilityChanged: \_ -> pure unit
+      } msg
 
-      Removed item a -> do
-        st <- H.get
-        H.modify_ _ { selected = filter ((/=) item) st.selected }
-        newSt <- H.get
-        let newItems = difference newSt.items newSt.selected
-        _ <- H.query unit $ Select.replaceItems newItems
-        pure a
+    Remove item -> do
+      newSt <- H.modify \oldSt -> oldSt { selected = filter ((/=) item) oldSt.selected }
+      void $ H.query _select unit $ H.tell $ Select.ReplaceItems $ difference newSt.items newSt.selected
```

Comment:
If we make `State` extensible, then this action could just be an extension of `Select` and the required state (the selected items list) could also just be an extension. Then there's no wrapper component required at all -- you have a typeahead that's still just the `Select` component.